sklearn/neighbors/classification.py

```diff
@@ -155,18 +156,39 @@ def predict(self, X):
         weights = _get_weights(neigh_dist, self.weights)
 
         y_pred = np.empty((n_samples, n_outputs), dtype=classes_[0].dtype)
```

Comment:
you shouldn't be constructing this dense output array. this defies the purpose of using a sparse (low-memory) structure.